Tests/SwiftFusionTests/Inference/ValuesStorageTests.swift

```diff
@@ -19,73 +19,55 @@ import XCTest
 import PenguinStructures
 @testable import SwiftFusion
 
-/// Test helpers.
-extension ArrayStorageImplementation {
-  /// Creates an array with `count` copies of `element`.
-  static func create(repeating element: Element, count: Int) -> Self {
-    let r = Self.create(minimumCapacity: count)
-    for _ in 0..<count { _ = r.append(element) }
-    return r
-  }
-  
-  /// Returns the element at `index`.
-  subscript(index: Int) -> Element {
-    return withUnsafeMutableBufferPointer { buffer in
-      return buffer[index]
-    }
-  }
-}
-
 class ValuesStorageTests: XCTestCase {
   
   func testDifferentiableMove() {
-    let values = DifferentiableArrayStorage.create(repeating: Pose2(0, 0, 0), count: 5)
-    let directions = VectorArrayStorage.create(repeating: Vector3(1, 0, 0), count: 5)
-    directions.withUnsafeMutableRawBufferPointer { directionBuffer in
-      values.move(along: UnsafeRawPointer(directionBuffer.baseAddress!))
+    var values = ArrayBuffer<DifferentiableArrayStorage>((0..<5).map { _ in Pose2(0, 0, 0) })
+    let directions = ArrayBuffer<VectorArrayStorage>((0..<5).map { _ in Vector3(1, 0, 0) })
+    directions.withUnsafeBufferPointer { directionsBuffer in
+      values.move(along: directionsBuffer.baseAddress!)
```

Comment:
I find myself a little surprised to see an unsafe API being used here.  Why doesn't `move(along:)` just accept any collection of whatever `directions`' `Element` type is?  And since `ArrayBuffer` is a Collection, why use `withUnsafeBufferPointer`?

Reply:
yes this improves things a lot! done.